src/lib.rs

```diff
@@ -23,29 +23,27 @@ use helper::docs::get_docs_help;
 use std::io::Write;
 
 /// Runs `halp`.
-pub fn run<Output: Write>(mut cli_args: CliArgs, output: &mut Output) -> Result<()> {
-    let config = if let Some(config_file) = cli_args
+pub fn run<Output: Write>(cli_args: CliArgs, output: &mut Output) -> Result<()> {
+    let mut config = if let Some(config_file) = cli_args
         .config
         .to_owned()
         .or_else(Config::get_default_location)
     {
-        let config = Config::parse(&config_file)?;
-        config.update_args(&mut cli_args);
-        Some(config)
+        Config::parse(&config_file)?
     } else {
-        None
+        let config = Config::default();
+        // write the default config into the disk & ignore the errors 'cause we don't want to crash
+        // the program if we can't write the config
```

Comment:
Maybe remove this comment and add a print statement so that user will know what the error is.

Reply:
Ok this's more reasonable, The reason that makes me choose this way it's that it might be annoying for some users who don't care about configs